src/test/scala/cromwell/CallCachingWorkflowSpec.scala

```diff
@@ -0,0 +1,131 @@
+package cromwell
+
+import java.util.UUID
+
+import akka.testkit._
+import cromwell.binding.types.{WdlIntegerType, WdlStringType, WdlArrayType}
+import cromwell.binding.values.{WdlInteger, WdlFile, WdlArray, WdlString}
+import cromwell.CromwellSpec.DockerTest
+import cromwell.util.SampleWdl
+
+import scala.language.postfixOps
+
+class CallCachingWorkflowSpec extends CromwellTestkitSpec("CallCachingWorkflowSpec") {
+  def cacheHitMessageForCall(name: String) = s"Call Caching: Cache hit. Using UUID\\(.{8}\\):$name\\.*"
+
+  val expectedOutputs = Map(
+    "file_passing.a.out" -> WdlFile("out"),
+    "file_passing.a.out_interpolation" -> WdlFile("out"),
+    "file_passing.a.contents" -> WdlString("foo bar baz"),
+    "file_passing.b.out" -> WdlFile("out"),
+    "file_passing.b.out_interpolation" -> WdlFile("out"),
+    "file_passing.b.contents" -> WdlString("foo bar baz")
+  )
+
+  "A workflow which is run twice" should {
+    "use cached calls on the second run" in {
+
+      /** This workflow has two identical calls in it (run in serial)
+        *
+        * Therefore the first invocation will get ONE call cache hit to itself
```

Comment:
Oh I get it, so 'b' hits the cache from 'a'...

This makes way more sense now.